osquery/sql/virtual_table.cpp

```diff
@@ -227,25 +227,28 @@ static int xFilter(sqlite3_vtab_cursor *pVtabCursor,
 
   // Now organize the response rows by column instead of row.
   auto &data = pVtab->content->data;
-  for (auto &row : response) {
+  auto row = response.rbegin();
+  while (row != response.rend()) {
```

Comment:
When iterating backwards using a `deque` we can push into the front and remove from the source back. Removing from the source when iterating forward causes a `n^2` runtime as each remove rescans the items.

Anecdotally we can show a 5x speed up when changing to a deque and removing from the end. Removing in the first place limits query time heap.